evennia/contrib/aws-s3-cdn.py

```diff
@@ -0,0 +1,847 @@
+"""
+This plugin migrates the Web-based portion of Evennia,
+namely images, javascript, and other items located
+inside staticfiles into Amazon AWS (S3) for hosting.
+
+INSTALLATION:
+
+1) If you don't have an AWS S3 account, you should create one now.
+
+Credentials required are an AWS IAM Access Key and Secret Keys,
+which can be generated/found in the AWS Console.
+
+Example IAM Control Policy Permissions, if desired:
+
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "evennia",
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
+                "s3:GetObjectAcl",
+                "s3:GetObject",
+                "s3:ListBucket",
+                "s3:DeleteObject",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": [
+                "arn:aws:s3:::YOUR_BUCKET_NAME/*",
+                "arn:aws:s3:::YOUR_BUCKET_NAME"
+            ]
+        }
+    ],
+    [
+      {
+         "Sid":"evennia",
+         "Effect":"Allow",
+         "Action":[
+            "s3:CreateBucket",
+         ],
+         "Resource":[
+            "arn:aws:s3:::*"
+         ]
+       }
+    ]
+}
+
+Advanced Users: The second IAM statement, CreateBucket, is only needed
+for initial installation. You can remove it later, or you can
+create the bucket and set the ACL yourself before you continue.
+
+2) This package requires the dependency "boto3," the official
+AWS python package. You can install it with 'pip install boto3'
+while inside your evennia virtual environment (or, simply
+in your shell if you don't use a virtual environment).
+
+3) Customize the variables defined below in secret_settings.py,
+then run 'evennia stop', 'evennia start', 'evennia collectstatic'
+
+AWS_ACCESS_KEY_ID = 'EUHUB20BU08AEU7' # CHANGE ME!
+AWS_SECRET_ACCESS_KEY = 'a/uoexauodabuq4j;kmw;kvka0d2' # CHANGE ME!
+AWS_STORAGE_BUCKET_NAME = 'mygame-evennia' # CHANGE ME!
+AWS_S3_REGION_NAME = 'us-east-1' # N. Virginia
+AWS_S3_OBJECT_PARAMETERS = { 'Expires': 'Thu, 31 Dec 2099 20:00:00 GMT', 'CacheControl': 'max-age=94608000', }
+AWS_DEFAULT_ACL = 'public-read'
+AWS_S3_CUSTOM_DOMAIN = '%s.s3.amazonaws.com' % settings.AWS_BUCKET_NAME
+AWS_AUTO_CREATE_BUCKET = True
+STATICFILES_STORAGE = 'evennia.contrib.aws-s3-cdn.S3Boto3Storage'
+You may also store these as environment variables of the same name.
+
+UNINSTALLATION:
+
+If you haven't made changes to your static files (uploaded images, etc),
+you can simply remove the lines you added to secret_settings.py. If you
+have made changes and want to install at a later date, you can export
+your files from your S3 bucket and put them in /static/ in the evennia
+directory.
+
+LICENSE:
+
+aws-s3-cdn contrib is (c) 2020, trhr and released under BSD 3-Clause
```

Comment:
Gotcha. The vast majority of the code inside the included classes were from django-storages and this is their attached license. Licenses seem compatible on their face, but IANAL and yours was called the "BSD License" and the existing code I used was "BSD 3-Clause License" so I included for clarity. I guess the bottom line is we can delete all the license information and call them identical?

Reply:
If the code and license is from Django storages, then it must indeed be included, but moreso their original copyright statement / author must be retained - that is after all the only main requirement of any BSD license. You should add yourself as author of the contrib (see how others do it in the contrib/ folder) - you will then fall under Evennia's general license. But you cannot add your own (c) into Django storages license nor modify their original license text.

Otherwise yes - it's compatible since it's the same license. It's just two copyright owners that need to be retained.

Reply:
I copied their authors.md, but pulled the full license text since it's the same license as yours, and just noted such.

For the test packages, I guess the way to go is just to grab the existing tests from django-storages, which looks fairly robust:
https://github.com/jschneier/django-storages/blob/master/tests/test_s3boto3.py